Exception in _handleCardChanged

uristrimber opened the pull request:
Fix: Revert card parsing logic in _handleCardChanged

This PR reverts the changes made to the input parsing logic in card_field.dart. The arguments map does not contain a nested 'card' key inside it.

Instead of incorrectly attempting to extract map['card'], this change reverts to passing the entire map directly to CardFieldInputDetails.fromJson.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved the Stripe card input widget to more robustly handle varying payment data formats, reducing errors and improving stability when users enter or update card details.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

uristrimber committed:
Update card_field.dart

uristrimber committed:
Rename project from stripe_workspace to flutter_stripe_fork

uristrimber committed:
Rename package from flutter_stripe_fork to stripe_workspace

remonh87 commented on packages/stripe/lib/src/widgets/card_field.dart:
```diff
@@ -541,9 +541,7 @@ class _MethodChannelCardFieldState extends State<_MethodChannelCardField>
     try {
       final map = Map<String, dynamic>.from(arguments);
-      final update = CardFieldInputDetails.fromJson(
-        Map<String, dynamic>.from(map['card']),
```
I tested both android and ios and for the ui works. Can you give me some more details of the version of the package that you are using?

uristrimber replied:
card is not a valid value in the map

remonh87 replied:
@uristrimber  when do you think you can make the changes. Would like to integrate it and take it along next version

uristrimber replied:
@remonh87 updated in f2f506c — now checks for `map['card']` and falls through to the flat map when it isn't there, so both payload shapes parse correctly. Ready for another look.

uristrimber replied:
My AI code-checker wrote this, what do you think @remonh87?

consider applying the same fix to card_form_field.dart.

The nested is Map check handles both payload shapes safely:

Wrapped {card: {...}} → parses nested map.
Flat {...} (or missing card key / non-Map value) → falls back to the outer map, avoiding the Null is not a subtype of Map crash reported in this issue.
null is Map is false, so a missing card key gracefully hits the fallback.
One follow-up: packages/stripe/lib/src/widgets/card_form_field.dart (_handleCardChanged, lines ~505–527) still unconditionally does Map<String, dynamic>.from(map['card']) and would crash the same way if the platform ever emits a flat payload for the card form. Worth mirroring this defensive check there so the two widgets behave consistently.

uristrimber committed:
Handle both wrapped and flat card change payloads

Native layers send {card: {...}} for some card states and the flat
fields for others. Extract map['card'] when present and fall through
to the outer map otherwise so incomplete-card events stop crashing.

uristrimber committed:
Merge branch 'main' into dev

remonh87 reviewed (approved):
looks good to me
